tsdb/engine/tsm1/reader.go

```diff
@@ -236,6 +238,18 @@ func (t *TSMReader) Close() error {
 	return t.accessor.close()
 }
 
+func (t *TSMReader) ValidateChecksum() map[string]map[string]bool {
```

Comment:
The problem is to use it i would need to expose ```fileAccessor``` OR I would have to move the WHOLE logic to the inspect_tool, which kills reusability and if you ever need this again somewhere you'd have to reimplement it.